packages/cli/lib/util/Configuration.ts

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2020-2023 Delft University of Technology and SynTest contributors
+ *
+ * This file is part of SynTest Framework - SynTest Core Graphing Plugin.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import Yargs = require("yargs");
+
+export type GeneralOptions = {
+  config: string;
+  applications: string[];
+  plugins: string[];
+  userInterface: string;
+};
+
+export type StorageOptions = {
+  logDirectory: string;
+  syntestDirectory: string;
+  tempSyntestDirectory: string;
+};
+
+export type LoggingOptions = {
+  consoleLogLevel: string;
+  logToFile: string[];
+};
+
+export type BaseOptions = GeneralOptions & StorageOptions & LoggingOptions;
+
+export class Configuration {
+  static configureUsage() {
+    return (
+      Yargs.usage(`Usage: syntest <command> [options]`)
+        // TODO examples
+        .epilog("visit https://syntest.org for more documentation")
+    );
+  }
+
+  static configureOptions(yargs: Yargs.Argv) {
+    return (
+      yargs
+        .option("config", {
+          alias: ["c"],
+          default: ".syntest.json",
+          description: "The syntest configuration file",
+          group: "General options:",
```

Comment:
We should probably make these group names constants or enum. At some point, a type is going to happen.

Reply:
Ironic how you made a typo in the word typo, haha.
Fixed.